tests/main/preseed-snapd-snap/task.yaml

```diff
@@ -0,0 +1,105 @@
+summary: Check that preseeding of current ubuntu cloud image works with snapd
+  and core18 works.
+description: |
+  This test checks that preseeding of Ubuntu cloud images with snap-preseed
+  command works, up to the point where the image is ready to be booted.
+  The test assumes cloud image with a core and lxd snaps in its seeds/.
+
+systems: [ubuntu-20.04-*]
+
+environment:
+  IMAGE_MOUNTPOINT: /mnt/cloudimg
+
+prepare: |
+  # the get_image_url_for_nested_vm is a convenient helper that returns
+  # a cloud image url matching current $SPREAD_SYSTEM.
+  #shellcheck source=tests/lib/nested.sh
+  . "$TESTSLIB/nested.sh"
+  wget "$(get_image_url_for_nested_vm)" -O cloudimg.img
+  mkdir -p "$IMAGE_MOUNTPOINT"
+
+restore: |
+  # any of the restore commands can fail depending on where execute part stopped,
```

Comment:
This needs `preseed.sh` to work and `|| true` has masked the failure.

Reply:
Good catch, fixed.